archetype-module-platform/src/main/resources/archetype-resources/pom.xml

```diff
@@ -95,12 +153,54 @@
 						</execution>
 					</executions>
 				</plugin>
+				<!-- Maven Compiler Plugin -->
+				<plugin>
+					<groupId>org.apache.maven.plugins</groupId>
+					<artifactId>maven-compiler-plugin</artifactId>
+					<version>3.11.0</version>
+					<configuration>
+						<source>${maven.compiler.source}</source>
+						<target>${maven.compiler.target}</target>
+					</configuration>
+				</plugin>
+				<!-- Maven Enforcer Plugin -->
+				<plugin>
+					<groupId>org.apache.maven.plugins</groupId>
+					<artifactId>maven-enforcer-plugin</artifactId>
+					<version>3.4.1</version>
+					<executions>
+						<execution>
+							<id>enforce-java</id>
+							<goals>
+								<goal>enforce</goal>
+							</goals>
+							<configuration>
+								<rules>
+									<requireMavenVersion>
+										<version>[3.6.0,)</version>
+									</requireMavenVersion>
+									<requireJavaVersion>
+										<version>[1.8,22)</version> <!-- Allows 8, 11, 17, 21 -->
+										<message>Java 8, 11, 17, or 21 is required. Other versions are not supported.</message>
```

Comment:
Why are you limiting it to 21 yet the module compiles well even on Java 24?

Reply:
fixed